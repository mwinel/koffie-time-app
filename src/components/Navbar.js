import React, { Component } from "react";
import { Link } from "react-router-dom";
import { Navbar as NavBar, Nav } from "react-bootstrap";
import { NavbarBrand } from "../styles/NavbarBrand";
import { ButtonPrimary } from "../styles/ButtonPrimary";
import logo from "../logo.png";

class Navbar extends Component {
  render() {
    return (
      <NavBar
        className="navbar navbar-expand-lg py-3"
        collapseOnSelect
        scrolling="true"
        expand="md"
        sticky="top"
      >
        <div className="container">
          <NavbarBrand>
            <Link to={"/"}>
              <img src={logo} width={40} alt="logo" />
            </Link>
          </NavbarBrand>
          <NavBar.Toggle aria-controls="responsive-navbar-nav" />
          <NavBar.Collapse id="responsive-navbar-nav">
            <ul className="navbar-nav mr-auto mt-2 mt-lg-0 ml-lg-5">
              <Nav.Link className="nav-link px-lg-3">Engineering</Nav.Link>
              <Nav.Link className="nav-link px-lg-3">Design</Nav.Link>
              <Nav.Link className="nav-link px-lg-3">Career</Nav.Link>
              <Nav.Link className="nav-link px-lg-3">Tutorials</Nav.Link>
            </ul>
            <ul className="navbar-nav mt-2 mt-lg-0 ml-lg-5">
              <Nav.Link className="nav-link px-lg-3">Login</Nav.Link>
            </ul>
            <ButtonPrimary className="ml-lg-4 mt-3 mt-lg-0">
              Get Started
            </ButtonPrimary>
          </NavBar.Collapse>
        </div>
      </NavBar>
    );
  }
}

export default Navbar;
