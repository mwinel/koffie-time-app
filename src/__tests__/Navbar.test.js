import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import { BrowserRouter } from "react-router-dom";
import Navbar from "../components/Navbar";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders navbar component", () => {
  act(() => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
      container
    );
  });
  expect(container.innerHTML).toBeDefined();
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
