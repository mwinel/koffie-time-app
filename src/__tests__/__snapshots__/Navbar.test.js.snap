// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navbar component 1`] = `
"<nav scrolling=\\"true\\" class=\\"navbar navbar-expand-lg py-3 navbar navbar-expand-md navbar-light sticky-top\\">
  <div class=\\"container\\">
    <nav class=\\"sc-bdVaJa gRvjIB\\"><a href=\\"/\\"><img src=\\"logo.png\\" width=\\"40\\" alt=\\"logo\\"></a></nav><button aria-controls=\\"responsive-navbar-nav\\" type=\\"button\\" aria-label=\\"Toggle navigation\\" class=\\"navbar-toggler collapsed\\"><span class=\\"navbar-toggler-icon\\"></span></button>
    <div class=\\"navbar-collapse collapse\\" id=\\"responsive-navbar-nav\\">
      <ul class=\\"navbar-nav mr-auto mt-2 mt-lg-0 ml-lg-5\\"><a href=\\"#\\" class=\\"nav-link px-lg-3 nav-link\\" role=\\"button\\">Engineering</a><a href=\\"#\\" class=\\"nav-link px-lg-3 nav-link\\" role=\\"button\\">Design</a><a href=\\"#\\" class=\\"nav-link px-lg-3 nav-link\\" role=\\"button\\">Career</a><a href=\\"#\\" class=\\"nav-link px-lg-3 nav-link\\" role=\\"button\\">Tutorials</a></ul>
      <ul class=\\"navbar-nav mt-2 mt-lg-0 ml-lg-5\\"><a href=\\"#\\" class=\\"nav-link px-lg-3 nav-link\\" role=\\"button\\">Login</a></ul><button class=\\"sc-bwzfXH fVpSfU ml-lg-4 mt-3 mt-lg-0\\">Get Started</button>
    </div>
  </div>
</nav>"
`;
